README.md

```diff
@@ -1,173 +1,175 @@
+
+## Event Subscriptions & Callbacks
+
+After creating an Erdstall client, event handlers can be set up using `Erdstall.on()` and `Erdstall.once()`, like
+
+```ts
+erdClient.on("receipt", (rec) => {
+	console.log(`Transaction by ${rec.tx.sender}`);
+});
+```
+
+Once all event handlers are setup, first `await Erdstall.initialize()` to initialize the client, then `await Erdstall.subscribe()` to subscribe to all balance proofs and transaction receipts for all users, or filter to a specific user with address `who` by calling `Erdstall.subscribe(who)`. `ErdstallSession.subscribeSelf()` is a shortcut for subscribing a session to the own user.
+
+<details><summary>The following <code>EnclaveEvent</code> events exist (emitted event types are written in parentheses):</summary>
+
+-   **`"open"`** is triggered when the connection to Erdstall is established.
+    This event is triggered again if the Erdstall connection is lost and automatically re-established.
+    At startup, you can alternatively use `await client.initialize()` instead of setting up a handler for this event.
+
+-   **`"config"(ClientConfig)`** is triggered when the client configuration is received from the operator after establishing the connections.
+
+-   **`"close"`** is triggered when the connection to Erdstall is terminated due to any reason.
+
+-   **`"error"(string | Error)`** is triggered when an error occurres within the Erdstall client.
+
+-   **`"receipt"(TxReceipt)`** is triggered when a receipt is received from a transaction subscription.
+    This does not include transaction receipts directly received as a response when issuing transactions.
+    A transaction receipt contains the transaction itself and the changed balances of all affected parties.
+
+-   **`"proof"(BalanceProof)`** is triggered when a balance proof is received from a proof subscription.
+    These balance proofs are used for exiting the system in case the operator goes offline.
+
+-   **`"exitproof"(BalanceProof)`** is triggered when a balance proof, intended for exiting the Erdstall network, is received from a proof subscription.
+    This happens after a call to `Erdstall.exit()` or during `Erdstall.leave()`.
+
+-   **`"phaseshift"`** is triggered when a phase shift in the Erdstall network occured, i.e., the last block of an epoch got processed by the enclave.
+</details>
+
+<details><summary>The following <code>LedgerEvent</code> smart contract events exist:</summary>
+All on-chain events return a type of the same name as the event, so they are
+omitted after each event name in the following.
+
+-   **`"Deposited"`**: an asset got deposited into Erdstall on-chain.
+
+-   **`"Withdrawn"`**: assets got withdrawn from Erdstall on-chain.
+
+-   **`"Frozen"`**: the operator failed to respond to a challenge within the challenge period. The Erdstall system is now frozen forever, and funds must be withdrawn using the latest _sealed_ balance proof.
```

Comment:
Regarding the semantics of a **sealed** balance proof: Maybe we want a some _Details_/_FAQ_/_Glossary_ section at the bottom where things like a **sealed** balance proof are explained. There might be more which could be extracted into this extra section, any ideas/thoughts?

Reply:
I like the idea of a Glossary

Reply:
Decided to do a glossary as a separate issue, possibly not in the README file. https://github.com/perun-network/erdstall-ts-sdk/issues/122